docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,503 @@
+# MeshTrafficPermission
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4222
+
+## Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a MeshTrafficPermission.
+
+## Considered Options
+
+* Create a MeshTrafficPermission policy
+
+## Decision Outcome
+
+Chosen option: create a MeshTrafficPermission policy
+
+### Overview
+
+MeshTrafficPermission allows blocking unwanted incoming traffic for the service.
+There are might be several reasons why it could be useful to block unwanted traffic:
+* following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+* cut off access for clients that shouldn't use your service (legacy clients, clients from another zone, etc.)
+
+### Specification
+
+Top-level targetRef can have all available kinds:
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+MeshTrafficPermission is an inbound policy that's why it has a "from" array:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+      name: ...
+```
+
+Configuration for MeshTrafficPermission has a single field "action" with 2 values "ALLOW" or "DENY".
+So the overall specification looks like this:
+
+```yaml
+type: MeshTrafficPermission
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+    name: ...
+  from:
+    - targetRef:
+        kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+        name: ...
+      conf:
+        action: ALLOW|DENY
+```
+
+We don't want to mix mTLS configs (like permissive mTLS or TLS version) with MeshTrafficPermission. 
+The main reason is that mTLS settings are applied for the entire inbound, while permissions could be set for a group of clients. 
+
+### Envoy configuration
+
+Envoy has [RBAC Network Filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/rbac_filter) 
+that allows configuring permissions for the list of principals.
+
+**Principal** is an entity that can be authenticated.<br>
+**Permission** is access granted to the principal.  
+
+Each Kuma service has a generated certificate with URIs that identify service name and tags.
+For example "backend" service in "us-east" zone of version v1 can have the following list of URIs:
+* spiffe://mesh-1/backend
+* kuma://kuma.io/zone/us-east
+* kuma://version/v1
+
+If we want to grant any permissions to these instances we can create the following Network RBAC:
+
+```yaml
+action: ALLOW
+policies:
+  "backend-us-east-v1-allow-all":
```

Comment:
The policyName is always the MeshTrafficPermission resource name it comes from?

If yes you should document that in the MADR

Reply:
It's the name of the most specific MeshTrafficPermission for the DPP. Added this to MADR

Reply:
I thought we settled on always having `MeshTrafficPermission` no?

Reply:
Still seem to be here no?

Reply:
Missed this place, fixed now